store/postgres/src/notification_listener.rs

```diff
@@ -0,0 +1,148 @@
+use fallible_iterator::FallibleIterator;
+use postgres::notification::Notification;
+use postgres::{Connection, TlsMode};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Barrier};
+use std::thread;
+use std::time::Duration;
+
+use futures::sync::mpsc::{channel, Receiver};
+use graph::prelude::*;
+
+/// This newtype exists to make it hard to misuse the `NotificationListener` API in a way that
+/// could impact security.
+pub struct SafeChannelName(String);
+
+impl SafeChannelName {
+    /// Channel names must be valid Postgres SQL identifiers.
+    /// If a channel name is provided that is not a valid identifier,
+    /// then there is a security risk due to SQL injection.
+    /// Unfortunately, it is difficult to properly validate a channel name.
+    /// (A blacklist would have to include SQL keywords, for example)
+    ///
+    /// The caller of this method is promising that the supplied channel name
+    /// is a valid Postgres identifier and cannot be supplied (even partially)
+    /// by an attacker.
+    pub fn i_promise_this_is_safe(channel_name: impl Into<String>) -> Self {
```

Comment:
What about having a less scary version of this that takes `&'static str`, for when it's possible to rule out the user-supplied case?

Reply:
Ouuu we think the same way! I actually had that originally, but then I realized that we have a lot of things that are created at runtime but leak to the static lifetime, so it could be easy to use one of them.

Reply:
We don't have so many of those things (I can only of the block ingestor) but a great point anyways, it would not be a real guarantee.